src/an_sso.coffee

```diff
@@ -176,8 +178,25 @@ window.AsmodeeNet = (->
                 if hash.state && hash.state == state
                     settings.callback_signin_error(hash.error + ' : ' + hash.error_description.replace(/\+/g, ' '))
 
+    checkDisplayOptions = () ->
+        if Object.keys(settings.display_options).length > 0
+            if settings.display in ['touch', 'popup']
+                tmpopts = {noheader: false, nofooter: false, lnk2bt: false, leglnk: true}
+                for opt, val of settings.display_options
+                    unless opt in Object.keys(tmpopts)
+                        delete settings.display_options[opt]
+                if Object.keys(settings.display_options).length > 0
+                    settings.display_options = AsmodeeNet.extend tmpopts, settings.display_options
+            else
+                settings.display_options = {}
+        if settings.display == 'touch'
+            if Object.keys(settings.display_options).length == 0
+                settings.display_options = {noheader: true, nofooter: true, lnk2bt: true, leglnk: false}
+            settings.cancel_uri = settings.redirect_uri if settings.cancel_uri is null
```

Comment:
It may be a better idea to use `... if !settings.cancel_uri`: if the user passes an empty string, the `cancel_uri` will stay that way.

`settings.cancel_uri ?= settings.redirect_uri` might even do the trick.